src/wp-includes/post.php

```diff
@@ -6048,6 +6048,16 @@ function get_pages( $args = array() ) {
 		$query_args['posts_per_page'] = $number;
 	}
 
+	/**
+	 * Filters query arguments passed to WP_Query in get_pages.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param array $query_args  Array of arguments passed to WP_Query.
+	 * @param array $parsed_args Array of get_pages() arguments.
+	 */
+	$query_args = apply_filters( 'get_pages_query_args', $query_args, $parsed_args );
```

Comment:
+1 on adding a unit test for the filter. A simple test could be to create a dummy page, use this filter to override the `$query_args` to use just `ID` with that page's ID, and then assert that the returned result includes only that single page.

Basically any test in which the arguments you pass to `get_pages()` would give different query results than the query arguments you use to override via the new filter.